### PR TITLE
[func.d] move `overloadApply` to `funcsem.d`, remove `import dmd.errors`

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

**Comment by thewilsonator** [comment]
Failure is unrelated

File: compiler/src/dmd/funcsem.d [hunk 1]
```diff
@@ -3400,3 +3400,115 @@ private void checkPrintfScanfSignature(FuncDeclaration funcdecl, TypeFunction f,
...
+ */
+extern (D) int overloadApply(Dsymbol fstart, scope int delegate(Dsymbol) dg, Scope* sc = null)
+{
+    Dsymbols visited;
+
+    int overloadApplyRecurse(Dsymbol fstart, scope int delegate(Dsymbol) dg, Scope* sc)
+    {
+        // Detect cyclic calls.
+        if (visited.contains(fstart))
+        return 0;
```
Comment (dkorpel) [inline compiler/src/dmd/funcsem.d:3426]: Indentation is off throughout this function

Comment (thewilsonator) [reply compiler/src/dmd/funcsem.d:3426]: I think my rebase messed that up somehow.

Comment (thewilsonator) [reply compiler/src/dmd/funcsem.d]: Fixed

**Review by dkorpel — APPROVED** [review]

**Pull request merged**